Core.Collectors/Web/HttpUtility.cs

```diff
@@ -19,6 +22,22 @@ public static async Task<JObject> ParseAsJObjectAsync(HttpResponseMessage respon
             return JObject.Load(jsonTextReader);
         }
 
+        public static async Task<Tuple<bool, JObject>> TryParseAsJObjectAsync(HttpResponseMessage response)
+        {
+            using Stream responseStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+            // Loading a stream / text into JObject expands ~10x in memory. In other words, a 100MB response, will approx. take 1GB ram.
+            // Our machines (where this is a problem --- for CG collection) have 16 GB ram. I am leaving 4 GB for OS (might not be needed but just in case) and the rest of the code.
+            // That leaves 12 GB to be shared between 6, or 2 GB each, so ~200 MB raw response.
+            if (responseStream.Length > 200 * MB)
```

Comment:
Did you confirm that this actually works? Is the response always fully buffered and the `Length` property valid and correct? If it's not fully buffered, and it's a streaming response without a Content-Length header, what is the value here?
0, -1, int.MaxValue, some arbitrary buffer size?

Reply:
The response seems to be always buffered. I think we can go with this.